src/UID2.Client/UID2Encryption.cs

```diff
@@ -513,16 +532,28 @@ private static IdentityScope DecodeIdentityScopeV3(byte value)
             return (IdentityScope)((value >> 4) & 1);
         }
 
-        private static IdentityType GetIdentityType(byte[] encryptedId)
+        private static bool ValidIdentityScopeAndType(byte value)
+        {
+            // 4th bit == 1 then EUID otherwise UID2
+            var identityScope = value >> 4;
+            // 2nd bit == 1 then Phone else Email
+            var identityType = (value & 0b1111) >> 2;
```

Comment:
Can we call `DecodeIdentityType` instead?

Reply:
So in my mind this method ensures there are no extra bits set, this will fail if there are other random bits set. Right now the decode identity scope and type methods only looks at the relevant bits 

I can combine them all together into something like TryDecodeIdentityScopeAndType(prefix, out IdentityType, out Identity Scope)

Reply:
can we ensure the logic of what bits go where isn't duplicated?

Reply:
Tried my best here

I can't seem to apply the same validation to encrypted data

So I've ended up with two methods DecodeEncryptedDataIdentityScopeV3 and TryDecodeIdentityScopeAndTypeV3